### PR TITLE
Implement a playerpanel

### PR DESCRIPTION
**Pull request by nikthechampiongr**
## About the PR

This PR implements a player panel which provides general information and actions for a player.

The information and screenshot can be seen on the screenshot as they are self explanatory.

Shared connections is how many other people are connected to the server with the same ip.

## Why / Balance

It is a convenience feature that allows admins to access important information about a player in 1 place as well as perform quick actions.

## Technical details

The playerpanel is a bui. Upon opening it, it will receive info the admin can access and display the appropriate information and enable the appropriate buttons.

Most buttons implement their functionality by running commands with the target player's guid.

The exceptions to this:
- The kick button which can only work with a username
- The toggle freeze button which sends a message to the server for the freeze to be performed.

The whitelist commands were also altered to also receive a guid.

Also a small quirk, the whitelist toggle will function immediately without expecting a message from the server. This is because we already know the user is able to use the command, and even if for some reason a player is whitelisted/dewhitelisted before we press the button the command that is executed will have no effect.

Rolebans are not counted just as what is returned from the db. Due to the way they are created, role bans that involve multiple roles are in reality multiple individual rolebans. The only way I found to actually differentiate between them is to compare the note creation time which appears to be the same across all individual rolebans created from a multi one. This is horrible but it's the only way I can see to do this properly.

## Media

image

- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: 
ADMIN:

- add: The player panel has now been added and can be invoked with the playerpanel command. It will display general information on the selected player, as well as allow for general actions to be performed on them.

### CODE REVIEW

**Comment by nikthechampiongr** [comment]
Well feedback round over, should be ready.

**Comment by nikthechampiongr** [comment]
Heisentest my beloved

**Comment by nikthechampiongr** [comment]
Rolebans cause me suffering.

File: Content.Client/Administration/UI/PlayerPanel/PlayerPanel.xaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    xmlns:ui="clr-namespace:Content.Client.UserInterface.Controls"
+    Title="{Loc ban-panel-title}" MinSize="300 300">
+    <BoxContainer Orientation="Vertical">
+        <BoxContainer Orientation="Horizontal">
+        <Label Name="PlayerName"/>
+        <Button Name="UsernameCopyButton" Text="{Loc player-panel-copy-username}"/>
+        </BoxContainer>
+       <BoxContainer Orientation="Horizontal">
+           <Label Name="Whitelisted"/>
+           <controls:ConfirmButton Name="WhitelistToggle" Text="False" Visible="False"></controls:ConfirmButton>
```
Comment (metalgearsloth) [inline Content.Client/Administration/UI/PlayerPanel/PlayerPanel.xaml:13]: Local plx.

File: Content.Server/Administration/Commands/PlayerPanelCommand.cs [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+        ui.SetPlayerState();
+    }
+
+    public override CompletionResult GetCompletion(IConsoleShell shell, string[] args)
+    {
+        if (args.Length == 1)
+        {
+            var options = _players.Sessions.OrderBy(c => c.Name).Select(c => c.Name).ToArray();
+
+            return CompletionResult.FromHintOptions(options, "<PlayerIndex>");
```
Comment (metalgearsloth) [inline Content.Server/Administration/Commands/PlayerPanelCommand.cs:51]: localise.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]
Rest looks okay just EUIs being a PITA to write.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**